app/oz/grammar/lexical.nearley

```diff
@@ -285,6 +286,53 @@ lexical_float -> "~":? [0-9]:+ "." [0-9]:* (("e" | "E") "~":? [0-9]:+):? {%
 %}
 
 ##############################################################################
+# List
+##############################################################################
+
+lexical_list ->
+    empty_list {% id %}
+  | particular_list {% id %}
+
```

Comment:
I think this is too complex because you are trying to convert to records twice, one at the `list_items` level and again at the `particular_list` rule. I think that instead, at the `list_items` level, you should just concat each variable entry to a simple array, and then at the `particular_list` rule, convert that array into a `lexicalRecord` tree like what you did at the samples.

You can also check the string parsing function `lexicalString` for an example of a recursive algorithm to convert a simple list of items into it's tree record representation.

I really don't like what we are doing here with JSON stringification and decoding to check if we are at the end of the list. Again, this is quite hacky and can be solved through recursion.

Reply:
I also dislike the idea of the stringify. I agree on returns a list to a particular_list then i will iterate them to build the list. thanks.